dtfs-central-api/src/v1/controllers/portal/deal/update-deal.controller.js

```diff
@@ -174,14 +179,16 @@ const removeFacilityIdFromDeal = async (dealId, facilityId, user, routePath) =>
 
 exports.removeFacilityIdFromDeal = removeFacilityIdFromDeal;
 
-// eslint-disable-next-line consistent-return
 exports.updateDealPut = async (req, res) => {
-  if (ObjectId.isValid(req.params.id)) {
-    const dealId = req.params.id;
+  try {
+    if (!ObjectId.isValid(req.params.id)) {
+      return res.status(400).send({ status: 400, message: 'Invalid Deal Id' });
+    }
 
+    const dealId = req.params.id;
     const { user, dealUpdate } = req.body;
 
-    await findOneDeal(dealId, async (deal) => {
+    return findOneDeal(dealId, async (deal) => {
```

Comment:
Not `await`ing here means the try-catch won't catch any errors that happens in `findOneDeal` (or the callback) - is that deliberate?

Reply:
I modified the `findOneDeal` function in `dtfs-central-api/src/v1/controllers/portal/deal/get-deal.controller.js` to change line 55 from `const dealsCollection = await db.getCollection('deals');` to `const dealsCollection = await db.getCollection('deals').then(() => {throw new Error('test error');});` to simulate an error happening.

When I call `PUT /v1/portal/deals/507f191e810c19729de860ea` with the code in your current change, the log output is
```
/app/src/v1/controllers/portal/deal/get-deal.controller.js:56
  .then(() => {throw new Error('test error');});
                     ^
Error: test error
    at /app/src/v1/controllers/portal/deal/get-deal.controller.js:56:22
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async findOneDeal (/app/src/v1/controllers/portal/deal/get-deal.controller.js:55:27)
```
and the app crashes (the next log messages are the app restarting) and the result in Postman is a socket hang up:
![image](https://github.com/UK-Export-Finance/dtfs2/assets/44811960/4a303532-2db6-4d6f-abbd-f62062296fa2)

When I change line 191 in this file back to `await findOneDeal(dealId, async (deal) => {` (or to `return await findOneDeal(dealId, async (deal) => {`) and call `PUT /v1/portal/deals/507f191e810c19729de860ea` then the log output is
```
Unable to update deal Error: test error
    at /app/src/v1/controllers/portal/deal/get-deal.controller.js:55:77
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async findOneDeal (/app/src/v1/controllers/portal/deal/get-deal.controller.js:55:27)
    at async exports.updateDealPut (/app/src/v1/controllers/portal/deal/update-deal.controller.js:191:5)
```
and the app does not crash (there are no log messages after this) and the result in Postman is a 500 response: 
![image](https://github.com/UK-Export-Finance/dtfs2/assets/44811960/3f72c127-8f49-46a1-9ce3-ccb7ed67d78f)

The difference is because the try-catch you've added can't catch errors from `findOneDeal` because we don't `await` it

Reply:
Have spoken and added `await` albeit this is not recommended and will need to be refactored.
The code refactorisation work is substantive and will be covered API refactorisation.